Code completion for expression builder [FEATURE]

m-kuhn opened the pull request:
Let the gif speak

![peek 2018-09-08 12-16](https://user-images.githubusercontent.com/588407/45253071-3bb94f80-b361-11e8-82e6-b1219bbe06bd.gif)

If someone else feels like jumping in and improve on this:

 * Add named parameters to more functions ;)
 * Implement context based things like `with_variable` or `@parent`
 * Think about function overloads
 * Convince QScintilla that on autocompletion of a function it should automatically open a bracket `(`
 * Quote field names on autocompletion

m-kuhn committed:
Code completion for expression builder [FEATURE]

m-kuhn committed:
Fix wordcharacters

m-kuhn committed:
Update some bits

nyalldawson commented on src/gui/qgscodeeditorexpression.h:
```diff
@@ -0,0 +1,95 @@
+  private:
+    void initializeLexer();
+    void updateApis();
+    QsciAPIs *mApis;
```
initialize to nullptr

nyalldawson reviewed (commented):
Nice!

m-kuhn committed:
Fix build warnings

m-kuhn committed:
Initialize to nullptr

m-kuhn committed:
Rename lexer

m-kuhn committed:
Expression callTips are lowercase

Expressions are case insensitive but in all the apis and samples lower case is used. Be conform.

3nids commented:
Love it!!!!
